### ISSUE TITLE
Adding support for HmIP-SLO

### ISSUE THREAD

**Issue by noxhirsch** [danielperna84/pyhomematic#168]
Hi,
I've just tried to add the new HmIP-SLO (outdoor brightness sensor). This is what I've done so far:

1. Added "IPBrightnessSensor" to homeassistant/components/homematic/__init__.py => `DISCOVER_SENSORS`

2. Added `"HmIP-SLO": IPBrightnessSensor` to `DEVICETYPES` in sensors.py

3. Added this to sensors.py:
```
class IPBrightnessSensor(HMSensor, HelperLowBatIP):
    """IP Sensor for outdoor brightness measure"""

    def __init__(self, device_description, proxy, resolveparamsets=False):
        super().__init__(device_description, proxy, resolveparamsets)

        # init metadata
        self.SENSORNODE.update({"CURRENT_ILLUMINATION": [1],
                                "AVERAGE_ILLUMINATION": [1],
                                "LOWEST_ILLUMINATION": [1],
                                "HIGHEST_ILLUMINATION": [1]})
```
These are the four values which are also shown in the HomeMatic UI. I tried to understand your code as much as possible and have done lots of trial and error - e.g. i added something like this for all values:
```
    def get_current_illumination(self, channel=None):
        return float(self.getSensorData("CURRENT_ILLUMINATION", channel))
```

It works halfways but very weird: I have two HmIP-SLOs, but each time I tried some different code in the IPBrightnessSensor class I get different sensors spit out "unavailable". So one time it's only one sensor and sometimes it's three or four. E.g. LOWEST_ILLUMINATION is available and gets updated while CURRENT_ILLUMINATION only ouputs "unavailable". As far as I can see, HomeMatic always sends the complete set of all sensor values (current, average, lowest, highest).

I also have the problem, that the unit_of_measurement is missing in HomeAssistant (it should be `lux`), so in history_graph I only get a "bar timeline" (I don't know the name) instead of a line chart. I've checked the code but found no way to add the unit.

Maybe you can help me here?

**Comment by noxhirsch** [comment]
Here's a picture of the discovered devices in HASS (the unit_of_measurement is added via customize.yaml as a workaround - I hope there's a better solution):
![slo_unavailable](https://user-images.githubusercontent.com/30938717/45717957-c9840e80-bb9b-11e8-8763-007d48e97a1b.JPG)

**Comment by danielperna84** [comment]
Do the sensors that are unavailable update to a correct state after a while when the illumination has changed? In general that behavior is new to me and doesn't make sense on first sight. Everything you've tried is looking correct so far.

**Comment by noxhirsch** [comment]
Very weird... I've rechecked the logs and can confirm that the sensor is getting updates as it should.
But now I noticed a `DUTY_CYCLE` error in the beginning:
```
ERROR (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: CURRENT_ILLUMINATION on XXXXXXXXXXXXXX:1 Exception: <Fault -1: 'Generic error (DUTY_CYCLE)'>
INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=UNREACH, value=True
```
I don't know why there should be a duty cycle error. The sensor is pushing the values to the CCU - so it can't be that I've requested too much values by restarting HomeAssistant often (or something else which could cause a request of the values). Besides that I get this error just for both CURRENT_ILLUMINATIONs everytime, but not for the other "sub-sensors" (especially not for the unavailable ones).

**Comment by noxhirsch** [comment]
Some seconds after the error I get an ordinary update - here's the "complete" log with device X & Y:
```
2018-09-19 00:13:35 ERROR (SyncWorker_2) [pyhomematic.devicetypes.generic] HMGeneric.getValue: CURRENT_ILLUMINATION on YYYYYYYYYYYYYY:1 Exception: <Fault -1: 'Generic error (DUTY_CYCLE)'>
2018-09-19 00:13:35 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=UNREACH, value=True
2018-09-19 00:13:35 ERROR (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on XXXXXXXXXXXXXX:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-19 00:13:36 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=UNREACH, value=True
2018-09-19 00:13:36 ERROR (SyncWorker_19) [pyhomematic.devicetypes.generic] HMGeneric.getValue: CURRENT_ILLUMINATION on XXXXXXXXXXXXXX:1 Exception: <Fault -1: 'Generic error (DUTY_CYCLE)'>
2018-09-19 00:13:36 ERROR (SyncWorker_17) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on YYYYYYYYYYYYYY:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-19 00:13:36 ERROR (SyncWorker_9) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on YYYYYYYYYYYYYY:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-19 00:13:36 ERROR (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on XXXXXXXXXXXXXX:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-19 00:13:36 ERROR (SyncWorker_9) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on YYYYYYYYYYYYYY:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=CONFIG_PENDING, value=False
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=DUTY_CYCLE, value=False
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=LOW_BAT, value=False
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=UNREACH, value=False
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=RSSI_DEVICE, value=-61
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE_STATUS, value=0
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE, value=3.0
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION, value=0.14
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION, value=0.13
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION, value=0.13
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION, value=0.13
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:44 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=CONFIG_PENDING, value=False
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=DUTY_CYCLE, value=False
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=LOW_BAT, value=False
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=UNREACH, value=False
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=RSSI_DEVICE, value=-80
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE_STATUS, value=0
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE, value=3.0
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION, value=1.58
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION, value=1.54
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION, value=1.56
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION, value=1.56
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION_STATUS, value=0
2018-09-19 00:13:49 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION_STATUS, value=0
```

**Comment by noxhirsch** [comment]
Then it would have to be the duty cycle of the CCU (and not only these sensors) and I should get errors with all devices? AFAIK the CCU is not able to contact battery powered sensors and has to wait until they connect to the CCU.
But of course sometimes some errors can trigger other weird errors - conveniently it's bedtime now so I'll check again tomorrow 😉

**Comment by danielperna84** [comment]
That's why I said it doesn't make sense that only a few sensors have that problem. But yes, as far as I know the CCU can also ask battery powered devices about their state. That depends on the device though. There are different operating modes to save battery power.
But yes, just look at it tomorrow to get a better picture if the duty cycle really is a relevant factor here.

**Comment by noxhirsch** [comment]
The sensors are still randomly unavailable. I don't know where the duty cycle error is coming from, but I think it has nothing (directly) to do with that. This morning (after a restart) all four sensors of one HmIP-SLO where unavailable while the others are completely available - but I've got an duty cycle error even for the working SLO.

**Comment by noxhirsch** [comment]
Now I've tried lots of stuff and checked the logs all the time .... but no idea what's causing this weird behavior. It seems like only restarting homeassistant is changing anything - reconnecting the CCU doesn't change the available sensors (neither does restarting the CCU).
What logs should I check in addition? At the moment I'm checking these:
```
    homeassistant.core: info
    pyhomematic.devicetypes.generic: info
```
Or do you have any idea what else I could try? I'm at my wits' end 😠

**Comment by noxhirsch** [comment]
I changed the pyhomematic loglevel to debug - but as you said, it doesn't look very useful to me. It seems like pyhomematic get's all the values as it should. This was before creating some temporary programs which seems to don't work either.
Maybe you can see something in the log I can't?
```
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: YYYYYYYYYYYYYY : {'TYPE': 'HmIP-SLO', 'SUBTYPE': 'SLO', 'ADDRESS': 'YYYYYYYYYYYYYY', 'RF_ADDRESS': 10380467, 'CHILDREN': ['YYYYYYYYYYYYYY:0', 'YYYYYYYYYYYYYY:1', 'YYYYYYYYYYYYYY:2', 'YYYYYYYYYYYYYY:3'], 'PARENT': '', 'PARENT_TYPE': '', 'INDEX': 0, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'SERVICE'], 'FIRMWARE': '1.0.16', 'AVAILABLE_FIRMWARE': '0.0.0', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': 'UP_TO_DATE', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': '', 'LINK_TARGET_ROLES': '', 'DIRECTION': 0, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 4}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: XXXXXXXXXXXXXX : {'TYPE': 'HmIP-SLO', 'SUBTYPE': 'SLO', 'ADDRESS': 'XXXXXXXXXXXXXX', 'RF_ADDRESS': 6454571, 'CHILDREN': ['XXXXXXXXXXXXXX:0', 'XXXXXXXXXXXXXX:1', 'XXXXXXXXXXXXXX:2', 'XXXXXXXXXXXXXX:3'], 'PARENT': '', 'PARENT_TYPE': '', 'INDEX': 0, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'SERVICE'], 'FIRMWARE': '1.0.16', 'AVAILABLE_FIRMWARE': '0.0.0', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': 'UP_TO_DATE', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': '', 'LINK_TARGET_ROLES': '', 'DIRECTION': 0, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 4}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: YYYYYYYYYYYYYY:0 : {'TYPE': 'MAINTENANCE', 'SUBTYPE': '', 'ADDRESS': 'YYYYYYYYYYYYYY:0', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'YYYYYYYYYYYYYY', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 0, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': '', 'LINK_TARGET_ROLES': '', 'DIRECTION': 0, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: YYYYYYYYYYYYYY:1 : {'TYPE': 'BRIGHTNESS_TRANSMITTER', 'SUBTYPE': '', 'ADDRESS': 'YYYYYYYYYYYYYY:1', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'YYYYYYYYYYYYYY', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 1, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': '', 'LINK_TARGET_ROLES': '', 'DIRECTION': 0, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: YYYYYYYYYYYYYY:2 : {'TYPE': 'COND_SWITCH_TRANSMITTER', 'SUBTYPE': '', 'ADDRESS': 'YYYYYYYYYYYYYY:2', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'YYYYYYYYYYYYYY', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 2, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'LINK', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': 'CONDITIONAL_SWITCH', 'LINK_TARGET_ROLES': '', 'DIRECTION': 1, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: YYYYYYYYYYYYYY:3 : {'TYPE': 'COND_SWITCH_TRANSMITTER', 'SUBTYPE': '', 'ADDRESS': 'YYYYYYYYYYYYYY:3', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'YYYYYYYYYYYYYY', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 3, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'LINK', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': 'CONDITIONAL_SWITCH', 'LINK_TARGET_ROLES': '', 'DIRECTION': 1, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: XXXXXXXXXXXXXX:0 : {'TYPE': 'MAINTENANCE', 'SUBTYPE': '', 'ADDRESS': 'XXXXXXXXXXXXXX:0', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'XXXXXXXXXXXXXX', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 0, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': '', 'LINK_TARGET_ROLES': '', 'DIRECTION': 0, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: XXXXXXXXXXXXXX:1 : {'TYPE': 'BRIGHTNESS_TRANSMITTER', 'SUBTYPE': '', 'ADDRESS': 'XXXXXXXXXXXXXX:1', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'XXXXXXXXXXXXXX', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 1, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': '', 'LINK_TARGET_ROLES': '', 'DIRECTION': 0, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: XXXXXXXXXXXXXX:2 : {'TYPE': 'COND_SWITCH_TRANSMITTER', 'SUBTYPE': '', 'ADDRESS': 'XXXXXXXXXXXXXX:2', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'XXXXXXXXXXXXXX', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 2, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'LINK', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': 'CONDITIONAL_SWITCH', 'LINK_TARGET_ROLES': '', 'DIRECTION': 1, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:37 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.__init__: device_description: XXXXXXXXXXXXXX:3 : {'TYPE': 'COND_SWITCH_TRANSMITTER', 'SUBTYPE': '', 'ADDRESS': 'XXXXXXXXXXXXXX:3', 'RF_ADDRESS': 0, 'CHILDREN': [], 'PARENT': 'XXXXXXXXXXXXXX', 'PARENT_TYPE': 'HmIP-SLO', 'INDEX': 3, 'AES_ACTIVE': 1, 'PARAMSETS': ['MASTER', 'VALUES', 'LINK', 'SERVICE'], 'FIRMWARE': '', 'AVAILABLE_FIRMWARE': '', 'UPDATABLE': True, 'FIRMWARE_UPDATE_STATE': '', 'VERSION': 1, 'FLAGS': 1, 'LINK_SOURCE_ROLES': 'CONDITIONAL_SWITCH', 'LINK_TARGET_ROLES': '', 'DIRECTION': 1, 'GROUP': '', 'TEAM': '', 'TEAM_TAG': '', 'TEAM_CHANNELS': [], 'INTERFACE': '', 'ROAMING': 0, 'RX_MODE': 0}
2018-09-21 00:38:46 DEBUG (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:0', key = 'RSSI_PEER'
2018-09-21 00:38:46 ERROR (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on YYYYYYYYYYYYYY:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:46 DEBUG (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'LOW_BAT'
2018-09-21 00:38:46 DEBUG (SyncWorker_11) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:0', key = 'RSSI_PEER'
2018-09-21 00:38:46 ERROR (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on YYYYYYYYYYYYYY:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:46 DEBUG (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'HIGHEST_ILLUMINATION'
2018-09-21 00:38:46 DEBUG (SyncWorker_15) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:0', key = 'RSSI_PEER'
2018-09-21 00:38:46 DEBUG (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:0', key = 'RSSI_PEER'
2018-09-21 00:38:46 ERROR (SyncWorker_11) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on XXXXXXXXXXXXXX:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:46 DEBUG (SyncWorker_11) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'LOW_BAT'
2018-09-21 00:38:46 DEBUG (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:0', key = 'RSSI_PEER'
2018-09-21 00:38:46 ERROR (SyncWorker_15) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on XXXXXXXXXXXXXX:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:46 DEBUG (SyncWorker_15) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'LOW_BAT'
2018-09-21 00:38:46 ERROR (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on YYYYYYYYYYYYYY:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:46 DEBUG (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'LOW_BAT'
2018-09-21 00:38:46 ERROR (SyncWorker_11) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on XXXXXXXXXXXXXX:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:46 DEBUG (SyncWorker_11) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'CURRENT_ILLUMINATION'
2018-09-21 00:38:46 ERROR (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on XXXXXXXXXXXXXX:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:46 DEBUG (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'LOW_BAT'
2018-09-21 00:38:46 ERROR (SyncWorker_15) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on XXXXXXXXXXXXXX:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:46 DEBUG (SyncWorker_15) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'AVERAGE_ILLUMINATION'
2018-09-21 00:38:46 ERROR (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on YYYYYYYYYYYYYY:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:46 DEBUG (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'CURRENT_ILLUMINATION'
2018-09-21 00:38:46 DEBUG (SyncWorker_5) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:0', key = 'RSSI_PEER'
2018-09-21 00:38:46 DEBUG (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:0', key = 'RSSI_PEER'
2018-09-21 00:38:47 DEBUG (SyncWorker_14) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:0', key = 'RSSI_PEER'
2018-09-21 00:38:47 ERROR (SyncWorker_11) [pyhomematic.devicetypes.generic] HMGeneric.getValue: CURRENT_ILLUMINATION on XXXXXXXXXXXXXX:1 Exception: <Fault -1: 'Generic error (DUTY_CYCLE)'>
2018-09-21 00:38:47 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=UNREACH, value=True
2018-09-21 00:38:47 ERROR (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on XXXXXXXXXXXXXX:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:47 DEBUG (SyncWorker_18) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'LOWEST_ILLUMINATION'
2018-09-21 00:38:48 ERROR (SyncWorker_6) [pyhomematic.devicetypes.generic] HMGeneric.getValue: CURRENT_ILLUMINATION on YYYYYYYYYYYYYY:1 Exception: <Fault -1: 'Generic error (DUTY_CYCLE)'>
2018-09-21 00:38:48 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=UNREACH, value=True
2018-09-21 00:38:48 ERROR (SyncWorker_5) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on YYYYYYYYYYYYYY:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:48 DEBUG (SyncWorker_5) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'LOW_BAT'
2018-09-21 00:38:48 ERROR (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on XXXXXXXXXXXXXX:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:48 DEBUG (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'LOW_BAT'
2018-09-21 00:38:48 ERROR (SyncWorker_14) [pyhomematic.devicetypes.generic] HMGeneric.getValue: RSSI_PEER on YYYYYYYYYYYYYY:0 Exception: <Fault -5: 'Unknown Parameter value for value key: RSSI_PEER'>
2018-09-21 00:38:48 DEBUG (SyncWorker_14) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'LOW_BAT'
2018-09-21 00:38:48 ERROR (SyncWorker_5) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on YYYYYYYYYYYYYY:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:48 DEBUG (SyncWorker_5) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'AVERAGE_ILLUMINATION'
2018-09-21 00:38:48 ERROR (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on XXXXXXXXXXXXXX:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:48 DEBUG (SyncWorker_10) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'XXXXXXXXXXXXXX:1', key = 'HIGHEST_ILLUMINATION'
2018-09-21 00:38:48 ERROR (SyncWorker_14) [pyhomematic.devicetypes.generic] HMGeneric.getValue: LOW_BAT on YYYYYYYYYYYYYY:1 Exception: <Fault -5: 'Unknown Parameter for value key: LOW_BAT'>
2018-09-21 00:38:48 DEBUG (SyncWorker_14) [pyhomematic.devicetypes.generic] HMGeneric.getValue: address = 'YYYYYYYYYYYYYY:1', key = 'LOWEST_ILLUMINATION'
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=CONFIG_PENDING, value=False
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=DUTY_CYCLE, value=False
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=LOW_BAT, value=False
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=UNREACH, value=False
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=RSSI_DEVICE, value=-82
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE_STATUS, value=0
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE, value=3.0
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION, value=1.57
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.52>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.54>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION, value=1.51
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.52>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION, value=1.53
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.52>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION_STATUS, value=0
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.53>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION, value=1.54
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.53>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION_STATUS, value=0
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: 1.54>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.53>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION_STATUS, value=0
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: 1.54>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.53>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:39:08 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=XXXXXXXXXXXXXX:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION_STATUS, value=0
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness CURRENT_ILLUMINATION: 1.54>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness AVERAGE_ILLUMINATION: 1.53>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness LOWEST_ILLUMINATION: 1.51>> 
2018-09-21 00:39:08 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity wz_brightness HIGHEST_ILLUMINATION: 1.57>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=CONFIG_PENDING, value=False
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=DUTY_CYCLE, value=False
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=LOW_BAT, value=False
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=UNREACH, value=False
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=RSSI_DEVICE, value=-59
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE_STATUS, value=0
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:0, interface_id=homeassistant-ip, key=OPERATING_VOLTAGE, value=3.0
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION, value=0.04
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION, value=0.03
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION, value=0.03
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=HIGHEST_ILLUMINATION_STATUS, value=0
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION, value=0.03
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: False>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=LOWEST_ILLUMINATION_STATUS, value=0
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=CURRENT_ILLUMINATION_STATUS, value=0
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 INFO (Thread-3) [pyhomematic.devicetypes.generic] HMGeneric.event: address=YYYYYYYYYYYYYY:1, interface_id=homeassistant-ip, key=AVERAGE_ILLUMINATION_STATUS, value=0
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness HIGHEST_ILLUMINATION: 0.04>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness CURRENT_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness AVERAGE_ILLUMINATION: 0.03>> 
2018-09-21 00:40:31 DEBUG (Thread-3) [pyhomematic.devicetypes.generic] HMDevice.event: Using callback <bound method HMDevice._hm_event_callback of <Entity az_brightness LOWEST_ILLUMINATION: 0.03>> 
```
These are all log entries regarding to my two sensors. You can ignore the "false" value of `CURRENT_ILLUMINATION`. It seems like HomeAssistant is delivering cached values of the other sensors but current illumination always gets its first value with the next update from the device.

Tomorrow I'll check the homeassistant.core log with debug level if I can see something there.

**Comment by danielperna84** [comment]
Please remove the `HelperLowBatIP` from your class. It doesn't seem to work with that helper, so this just unnecessarily bloats the logs with failure-messages.
Besides that you can add the helper `HelperRssiDevice` with the latest pyhomematic version (0.1.49). Both of these changes won't fix the problem, but a lot of unnecessary error messages will be removed. Well, maybe with the new pyhomematic the lowbat-helper actuall works.

That being said, the events that arrive at Home Assistant (even for `CURRENT_ILLUMINATION` in this example) are looking good and should update the entities. All the way at the bottom of the loc all parameters have a numeric value. So even if not all values a fetched on startup, it looks like the will be updated over time.

**Comment by noxhirsch** [comment]
Oh, I thought `HelperLowBatIP` only produces failure messages for the unavailable devices (I didn't count them), because I got a "Battery: High" on the working devices. But I think that's because no value returns _false_ as like the device's low_bat is also false => so both get's converted to High.
I also wondered where the heck I saw `HelperRssiDevice`, because I couldn't use it and couldn't find it in helper.py. Now I know that I found it here on GitHub 😆 

Anyway - I think I found the reason (not the solution): When HomeAssistant is booting it's asking Homematic to update all the devices (as you wrote above). Homematic then tries to reach these two devices and thinks that they are unavailable for a short time (until they call the CCU themselves).
What I don't understand:
1. Why didn't I saw that earlier - I checked the CCU "Servicemeldungen" for exactly this behavior...
2. Why is HomeAssistant getting some of the values and some others not? (maybe Homematic is sending these values before "finding out" that the devices are offline. So the unavailable status changes while sending the values)
3. Why only the HmIP-SLOs and not e.g. the window/door sensors? (maybe an error in the firmware which make Homematic think that this device can answer calls from the CCU)
4. **Why doesn't HomeAssistant update the state later? In the homematic.core log you can see that HomeAssistant is only getting state changes for the available sensors. The unavailable sensors are only once in the log.**

Is there a solution for number 4?

**Comment by danielperna84** [comment]
I didn't count either. So technically it may still be correct. Not knowing much about the device it looked like it generally has some problems. So you may be right about that.

1. Is there something related to these devices in the "Servicemeldungen" when you get the error?
2. Basically what HA is doing on every restart:
```python
from xmlrpc.client import ServerProxy
p = ServerProxy("http://ccu:2010")
# for all devices
state = p.getValue("address:channel", "parameter")
```
If the `getValue` method on the CCU fails, then the devices for some reason the device is _unavailable_. It indeed does not make sense, that only a subset of paramters of a single device are not available. However, thinking about the duty cycle: if the device is broken and sends a lot of data, then maybe at the point where HA is asking for the value, the duty cycle of the device is already almost exhausted, only leaving roome for 1-2 replies, and then not replying for the remaining parameters.
3. Yes, it could be a firmware bug. Or some configuration issue. Maybe the device does something with "Direktverknüpfungen" in the background for whatever reason. Just like some devices create system variables on their own.
4. It should actually do that. I'll see if I can reproduce this by taking out the battery of some sensor and then plugging it back in.

**Comment by noxhirsch** [comment]
1. Yes, I get a "Gerätekommunikation gestört" (most times for both devices) which disappears when the device reports the new values after a while (in the usual updating cycle). If I remember correctly I only checked the "Servicemeldungen" _after_ restarting and expected a "Gerätekommunikation war gestört" which does not appear.
2. The value for duty_cycle of the device in the log is always false and a graph I created in the HomeMatic UI is perfectly fine without any holes. Even after a restart of HA everything is fine with the graph in HomeMatic UI, because the update cycle of the device stays the same.

Here's the cleaned log of `homeassistant.components.homematic: debug`:
```
2018-09-29 16:36:53 DEBUG (SyncWorker_17) [homeassistant.components.homematic] Subscribe channel 0 from az_brightness HIGHEST_ILLUMINATION
2018-09-29 16:36:53 DEBUG (SyncWorker_17) [homeassistant.components.homematic] Subscribe channel 1 from az_brightness HIGHEST_ILLUMINATION
2018-09-29 16:36:55 DEBUG (SyncWorker_19) [homeassistant.components.homematic] Subscribe channel 0 from az_brightness CURRENT_ILLUMINATION
2018-09-29 16:36:55 DEBUG (SyncWorker_19) [homeassistant.components.homematic] Subscribe channel 1 from az_brightness CURRENT_ILLUMINATION
2018-09-29 16:36:55 DEBUG (SyncWorker_6) [homeassistant.components.homematic] Subscribe channel 0 from az_brightness AVERAGE_ILLUMINATION
2018-09-29 16:36:55 DEBUG (SyncWorker_6) [homeassistant.components.homematic] Subscribe channel 1 from az_brightness AVERAGE_ILLUMINATION
2018-09-29 16:36:55 DEBUG (SyncWorker_17) [homeassistant.components.homematic] Subscribe channel 0 from az_brightness LOWEST_ILLUMINATION
2018-09-29 16:36:55 DEBUG (SyncWorker_17) [homeassistant.components.homematic] Subscribe channel 1 from az_brightness LOWEST_ILLUMINATION
[...]
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'HIGHEST_ILLUMINATION' value: 6922.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'HIGHEST_ILLUMINATION' value: 6922.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'HIGHEST_ILLUMINATION' value: 6922.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'HIGHEST_ILLUMINATION' value: 6922.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'LOWEST_ILLUMINATION' value: 6899.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'LOWEST_ILLUMINATION' value: 6899.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'LOWEST_ILLUMINATION' value: 6899.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'LOWEST_ILLUMINATION' value: 6899.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'AVERAGE_ILLUMINATION' value: 6912.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'AVERAGE_ILLUMINATION' value: 6912.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'AVERAGE_ILLUMINATION' value: 6912.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'AVERAGE_ILLUMINATION' value: 6912.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'HIGHEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'HIGHEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'HIGHEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'HIGHEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'CURRENT_ILLUMINATION' value: 6901.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'CURRENT_ILLUMINATION' value: 6901.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'CURRENT_ILLUMINATION' value: 6901.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'CURRENT_ILLUMINATION' value: 6901.0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'LOWEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'LOWEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'LOWEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'LOWEST_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'CURRENT_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'CURRENT_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'CURRENT_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'CURRENT_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness HIGHEST_ILLUMINATION received event 'AVERAGE_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness CURRENT_ILLUMINATION received event 'AVERAGE_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness AVERAGE_ILLUMINATION received event 'AVERAGE_ILLUMINATION_STATUS' value: 0
2018-09-29 16:37:28 DEBUG (Thread-3) [homeassistant.components.homematic] az_brightness LOWEST_ILLUMINATION received event 'AVERAGE_ILLUMINATION_STATUS' value: 0
```
This supports my theory, that the first values get sent to HA before HomeMatic thinks the device is unreachable. It's a little bit different, because there are no values at the beginning, but when HA subscribed to the sensors it subscribed first to highest and current illumination which are available in HA after this restart all the following sensors are offline.

There also seems to be a malfunction, because all values gets reported to all sensors - but I think this has nothing to do with the other problem I have.

**Comment by danielperna84** [comment]
> There also seems to be a malfunction, because all values gets reported to all sensors - but I think this has nothing to do with the other problem I have.

That's normal because it's the same device. All those entities share the same address. Otherwise general device info like RSSI or low battery would only be update on one of the entities. So all entities monitor the events and then decides if the content is relevant to them.

As for the rest: are the devices already mounted at their final position? I really think you shoould resolve the "Gerätekommunikation gestört" messages. Yes, we get some of the data. And yes, the values of the entities should update as the events arrive. But that's a generic issue right now. For the implementation of your device though you should make sure that everything is generally working without any errors. So maybe placing the device near the CCU because of poor reception could help.

**Comment by noxhirsch** [comment]
Okay, good to know 👍 

The devices are mounted in their final position, but I already tried using them aside from the CCU. Also the reception in final position is quite good (about -60dB which is still green in the new CCU firmware).
Every value from the device gets transmitted perfectly fine (without any "Gerätekommunikation gestört" message) - it's only the CCU which cannot ask for values and *thinks* that the communication is not working. But it is only a sensor and not an actor so the CCU should not try to contact the device directly. Even for saving changed device settings the CCU waits for the sensor without "Gerätekommunikation gestört" messages as it should be. 
So the only errors appear when requesting the actual values from the CCU. All the other time everything is working and the reception stays at nearly the same good level (I have devices with way worse reception working perfectly).
After lots of testing I opened an issue on OCCU's GitHub on Saturday about this behavior. But I don't know if and when they do something about that. I have no idea what else I can do.

**Comment by danielperna84** [comment]
I have opened a PR for HASS that should fix the problem with devices remaining in the unavailable state here: https://github.com/home-assistant/home-assistant/pull/17341
Please modify your local code and see if in general your code is working, not taking into account the connectivity issue. If the device starts working as events are received, then your code is working as it should and can be added to pyhomematic.

**Comment by noxhirsch** [comment]
Quick update: It works 👍 Thank you very much!
I'll add the code to pyhomematic for the device later.

**Comment by danielperna84** [comment]
Since your code for HASS has been merged and the new pyhomematic version is now public I'm closing this issue. Thanks for your work.
